test_pytest_timeout.py

```diff
@@ -482,3 +482,36 @@ def custom_trace(*args):
     module.custom_trace = custom_trace
 
     assert pytest_timeout.is_debugging(custom_trace)
+
+
+def test_logging_customization(testdir):
+    testdir.makepyfile(
+        conftest="""
+        import pytest
+
+        @pytest.hookimpl(hookwrapper=True)
+        def pytest_runtest_makereport(item, call):
+            r = yield
+            if not hasattr(r, "get_result"):
+                return
+            report = r.get_result()
+            timed_out = False
+            if hasattr(call.excinfo, "value"):
+                msg = getattr(call.excinfo.value, "msg", None)
+                if isinstance(msg, str) and msg.startswith("Timeout >"):
+                    timed_out = True
```

Comment:
What would you think of using `item.name` to know whether the test should have timed out or not?  Or even go further and just write two tests, or rather parametrise this test.

Reply:
I am not sure if the name thing would work: `test_suppresses_timeout_when_debugger_is_entered` contains the word `timeout` but the test asserts that no timeout occurs. `test_ini_method` does not contain the string `timeout` but the test expects a timeout. There is no correlation between the names and the expected outcome. Also, using `item.name` might break the test when another feature pr adds another test. Seeing an unrelated test fail might become an unnecessary nightmare for the developer of the other feature. The only possible way would be to create a decorator `@expect(timeout: Optional[bool])` and mark all tests. I am not sure if that is ideal. The `call.excinfo.value` method should be reliable and detect a timeout if it has occurred I think. Ideally, we could shift this conftest into a wrapper such that all tests are run with this test. That would probably also require a separate decorator for all existing tests though. What'd be your preference?
I'll experiment with it later this week and maybe I can come up with something more general...

Reply:
regarding names, your test only contains two test names: `test_times_out` and `test_does_not_time_out`.  The rest of the tests in this file are isolated from your test.

But I think it's even better to parametrise this test and make a different conftest.py based on the parameter so it knows whether to expect a timeout or not

Reply:
Okay, I'll try to push it into as many tests as I can. However, my schedule has changed a little so I had to postpone working on this to next week...

Reply:
I managed to get a wide-ranging implementation of the test for this pr working. It is very experimental let's say if not completely opaque and ugly.

It uses a `pytest` hook to create a `conftest.py` file before every test in the `pytest-timeout` suite is run. The thing that annoys me most is that I am not sure if this can be easily identified as the source of tests failing. Hence I marked that last commit as temporary; either it shall be discarded or possibly changed merged into the previous one.
Here is the commit message:

```
TEMPORARY commit: implement test for #90

This commit implements a test for #90 (logging customisation) that is
wide-ranging as it (ab)uses all other tests in the suite to also check
that the feature implemented in the above-mentioned pull request works.

As the implications are currently not completely clear, this commit is
marked as temporary and will either be discarded or merged into a
previous commit in the pr before it is merged.

The most important open question is:
If a change breaks #90, will this be visible to the developers?
The tests in the suite run other tests using the `tempdir` facility and
this hacky implementation just adds a `conftest.py` to each of the
temporary directories. Will the message from the assert message ever
reach the output of the pytest-timeout test suite? Can that be somehow
enforced from within the child-tests or their pytest runner?
```

Reply:
@flub: If you find the time, I'd be glad to hear your opinion about that idea.

Reply:
hi! again, thanks for your patience.
I don't think this is the right direction though, we shouldn't be changing the entire test suite, just adding a few tests for the new functionality.  I think your original test was fine as structure and overall approach, i thought the only tweak needed was to split it into two tests so you didn't have to try and detect whether the test was was going to time out or not.  Since both tests are rather similar i suggested parametrisation as a shortcut, but that's not strictly required if you're not sure how to use that.